### ISSUE TITLE
Permission already exists - Cannot approve new API permission requests

### ISSUE THREAD

**Issue by 5erban** [SharePoint/sp-dev-docs#4408]
#### Category
- [ ] Question
- [ ] Typo
- [x] Bug
- [ ] Additional article idea

#### Expected or Desired Behavior

When you try to approve API permissions, they get approved.

#### Observed Behavior

Error: Permission already exists

#### Steps to Reproduce

Same steps as in [#3823](https://github.com/SharePoint/sp-dev-docs/issues/3823). This is not fixed on our tenant.

**Comment by lahuey** [comment]
@5erban Can you email me at lahuey@microsoft.com? We can set up some time to look into your issue.

Sorry about preemptively closing your previous bug. I'll make sure this one stays open until things are resolved.

**Comment by 5erban** [comment]
@lahuey , no problem. Actually I found a way to fix it. We had two SharePoint Online Client Extensibility apps in AAD, one was from Microsoft and one was only listed when search for All Enterprise Apps. 

The SharePoint Online Client Extensibility App from Microsoft had a number of permissions already approved. However, `spfx3rdPartyServicePrincipalId` was referencing the  SharePoint Online Client Extensibility App listed under All apps.

I'm guessing that, at some point, the reference from Microsoft's SPOCE changed to All's SpOnCliExt. I'm not sure how this can happen.

Bottom line: you couldn't approve some permissions because they were already approved in the other SPOCE app.

The fix that worked for us: 
1. Delete both the app as well as the app registration for SharePoint Online Client Extensibility listed under All category in Enterprise Apps.
2. Redeploy a SPFx package - this automatically creates a new App Registration for SPOCE.

Removing all permissions from Microsoft's SPOCE did not work.

Please let me know if you need further info on this.

**Comment by lahuey** [comment]
Awesome! Deleting the app was actually the first thing I was going to recommend. We changed the design recently to put things in a good state if the application is deleted. Glad to know you're unblocked.

Thanks!
